lfs/config.go

```diff
@@ -416,20 +422,27 @@ func (c *Configuration) loadGitConfig() bool {
 	c.gitConfig = make(map[string]string)
 	c.extensions = make(map[string]Extension)
 	uniqRemotes := make(map[string]bool)
+	uniqInsteadOfs := make(map[string]string)
+	c.insteadOfs = make(map[string]string)
 
 	configFiles := []string{
 		filepath.Join(LocalWorkingDir, ".lfsconfig"),
 
 		// TODO: remove .gitconfig support for Git LFS v2.0 https://github.com/github/git-lfs/issues/839
 		filepath.Join(LocalWorkingDir, ".gitconfig"),
 	}
-	c.readGitConfigFromFiles(configFiles, 0, uniqRemotes)
 
 	listOutput, err := git.Config.List()
+
 	if err != nil {
 		panic(fmt.Errorf("Error listing git config: %s", err))
 	}
 
+	c.readGitInsteadOfConfig(listOutput, uniqInsteadOfs)
+	c.insteadOfs = uniqInsteadOfs
+
+	c.readGitConfigFromFiles(configFiles, 0, uniqRemotes)
```

Comment:
There's a silent dependency here on `readGitInsteadOfConfig` being called before `readGitConfigFromFiles`, and having assigned `c.insteadOfs` before it, even though that member var isn't used anywhere else. 

I think the call to `readGitInsteadOfConfig` should be nested *inside* the  `readGitConfigFromFiles` to make this explicit, and the output `uniqInsteadOfs` not stored in the `Config` at all, just used to affect the behaviour of the rest of `readGitConfigFromFiles`, as it currently does. This would eliminate the hidden 'magic' ordering and unused state.